docs/development/communication.md

```diff
@@ -1,11 +1,147 @@
+
+#### Message Layout
+
+There are two frame formats defined, the base frame format with 11 identifier bits and the extended frame format, using 29 bits for message identification [2].
+
+The (simplified) base frame format:
+
+| #Bits |  1  | 11 |  7  | 0 .. 64 |  16 |  2  |  7  |
+|-------|:---:|:--:|:---:|:-------:|:---:|:---:|:---:|
+|       | SOF | ID | CTL |   DATA  | CRC | ACK | EOF |
+
+**SOF:** Start of frame <br />
+**ID:** Unique Identifier which also determines the priority <br />
+**CTL:** Control bits, including the length of data (0..8 bytes) <br />
+**CRC:** Cyclic Redundancy Check, used to detect transmission errors <br />
+**ACK:** Acknowledgment bits <br />
+**EOF:** Bit sequence to signal the end of frame
+
+With the extended frame format, there are additionaly 18 bits used for the ID field which can be used to split the ID into message priority and target and source address.
+
+#### Arbitration
+
+Since CAN is masterless, every node on the bus can start sending at any time. To detect collisions, the transceiver always compares its output to the input. If another device with a higher priority (lower ID) is sending at the same time, a difference between input and output will be detected and the lower priority device will retreat.
+
+### UART
+
+The Universal Asynchronous Receiver Transmitter is the most common serial communication used in MCUs. It has a receive (RX) and a transmit (TX) line, which are cross-connected between two devices. It allows only two connected devices and does not need any additional wiring, both microcontrollers should share the same ground and the same maximum voltage.
+
+Since there are only two devices and separate wires for sending and receiving, no identification or arbitration is needed. UART does not implement any error checking, so higher layers need to check for data integrity.
+
+### Bluetooth
+
+Bluetooth (BT) is a widely used wireless connection protocol developed for medium-bandwidth, low range applications. Different versions are available for low-energy, high-speed or basic needs. For many application layer protocols, the Serial Port Profile is mostly suited as it simulates a serial interface and is therfore interchangeable with UART or USB
+
+### USB
+
+The Universal Serial Bus uses differential logic levels with two wires, just like CAN. It is implemented as a primary/secondary topology, meaning a host device initiates and organises communication to connected peripheric devices. Each devices has its own unique address and broadcasting is not possible. Since USB was developed for desktop use with high-bandwidth and low cable range, it is not widely used for MCU communication. It is however often used to connect a MCU with a PC via an USB-to-UART converter for programming and debugging. In this case, the same rules as for [UART](#UART) applies concerning error checking.
```

Comment:
```suggestion
The Universal Serial Bus uses differential logic levels with two wires, just like CAN. It is implemented as a primary/secondary topology, meaning a host device initiates and organizes communication to connected peripheral devices. Each devices has its own unique address and broadcasting is not possible. Since USB was developed for desktop use with high-bandwidth and low cable range, it is not widely used for MCU communication. It is however often used to connect a MCU with a PC via an USB-to-UART converter for programming and debugging. In this case, the same rules as for [UART](#UART) applies concerning error checking.
```